build.sbt

```diff
@@ -1,26 +1,43 @@
+import sbtcrossproject.CrossPlugin.autoImport.{ crossProject, CrossType }
 import ScalaModulePlugin._
 
-scalaModuleSettings
-scalaModuleSettingsJVM
-
-name := "scala-collection-contrib"
-version := "0.1.1-SNAPSHOT"
-
-crossScalaVersions in ThisBuild := Seq("2.13.0")
-
-scalacOptions ++= Seq("-opt-warnings", "-language:higherKinds", "-deprecation", "-feature", "-Xfatal-warnings")
-scalacOptions in (Compile, doc) ++= Seq("-implicits", "-groups")
-
-testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v", "-s", "-a")
-parallelExecution in Test := false  // why?
-
-mimaPreviousVersion := Some("0.1.0")
-
-homepage := Some(url("https://github.com/scala/scala-collection-contrib"))
-licenses := Seq("Apache-2.0" -> url("https://www.apache.org/licenses/LICENSE-2.0"))
-
-libraryDependencies ++= Seq(
-  "junit"            % "junit"           % "4.12"   % Test,
-  "com.novocode"     % "junit-interface" % "0.11"   % Test,
-  "org.openjdk.jol"  % "jol-core"        % "0.9"    % Test
-)
+lazy val root = project.in(file("."))
+  .aggregate(`scala-collection-contribJS`, `scala-collection-contribJVM`)
+  .settings(disablePublishing)
+
+lazy val `scala-collection-contrib`  = crossProject(JVMPlatform, JSPlatform)
```

Comment:
May I suggest using a `CrossType.Pure` project type?

Reply:
I tried, but encountered test compilation error during `root/test`, though fine with `scaa-collection-contribJ(S|VM)/test`.

Reply:
Fixed and now `Pure`.

Reply:
Weirdly, it seems that e15c87d has been overridden by another commit.

Reply:
Sorry, it was my mistake.
`test` or `root/test` still causes error as mentioned above, so removed the commit.

Reply:
Indeed, I’ve just tried and observed the same behavior as you. Any idea, @SethTisue or @sjrd? (I guess the problem might come from sbt-scala-module or sbt-crossproject?). In the meantime I’ve found a workaround (see exoego/scala-collection-contrib#1)

Reply:
Relying on aggregates with a Pure cross-project `in(file("."))` leads to issues, because the default root project created by sbt will *also* be `in(file("."))` and hence will try to pick up the sources and tests of the cross-project, leading to weird issues because the root project is not configured properly (for example it will be missing dependencies).

The solution is not to rely on aggregates. Rewire the default `test` command instead, or specify a project in which sbt should execute tasks by default that is different from the root project.

Reply:
Oh, right, I missed the fact that both projects were `in(file("."))`. We should just use a different root directory for the cross project, then.

Reply:
(I merged regardless, but of course a followup PR would be welcome.)